src/p2p_networking.rs

```diff
@@ -1,25 +1,50 @@
-use crate::errors::RuntimeError;
+use crate::async_runtime::Handle;
+use crate::errors::{LipaError, MapToLipaError, RuntimeError};
 use crate::{NodeAddress, PeerManager};
 use bitcoin::secp256k1::PublicKey;
-use log::debug;
+use log::{debug, error, trace};
+use std::fmt;
 use std::net::SocketAddr;
 use std::str::FromStr;
 use std::sync::Arc;
 use std::task::Poll;
 use std::time::Duration;
+use tokio::task::JoinHandle;
 use tokio::time::sleep;
 
 pub(crate) struct P2pConnections {}
 
 impl P2pConnections {
-    pub async fn connect_peer(
+    pub fn run_bg_connector(
```

Comment:
I would prefer `start_p2p_connector` or `start_p2p_background_connector`

Reply:
how about `init` instead of `start`? I took out the `p2p` part, because it is a function of the struct `P2pConnections`, so whenever you call it you'd run P2pConnections::start_p2p_background_connector which is somewhat redundant.

In fact I'd kind of propose the opposite, and removing both connection and p2p because of the redundancy and instead call it something like:
`P2pConnections::init_background_service()` or `P2pConnections::init_background_task()`.

Also, imo `P2pConnections` should be renamed to `P2pConnection`, because so far it is really only dealing with one connection, namely the connection with the LSP.

WDYT?

Reply:
> how about init instead of start? I took out the p2p part, because it is a function of the struct P2pConnections, so whenever you call it you'd run P2pConnections::start_p2p_background_connector which is somewhat redundant.

> In fact I'd kind of propose the opposite, and removing both connection and p2p because of the redundancy and instead call it something like:
> P2pConnections::init_background_service() or P2pConnections::init_background_task().

Agreed. `P2pConnections::init_background_service()` or `P2pConnections::init_background_task()` SGTM. 

> Also, imo P2pConnections should be renamed to P2pConnection, because so far it is really only dealing with one connection, namely the connection with the LSP.

I think we might quickly want to have more than one LSP node but for now yes, we can change to singular.

Reply:
Fixed: aa2f6c6ad837b014a771429f51d9cc5fa2db57de